### ISSUE TITLE
Do things work with 3D land masks?

### ISSUE THREAD

**Issue by rabernat** [ocean-eddy-cpt/gcm-filters#31]
In our [existing examples](https://gcm-filters.readthedocs.io/en/latest/tutorial_irregular_grid.html), we filter 3D data with a 2D wet mask. This is relatively simple because the wet mask has the same dimensions as the Laplacian operation.

But what if the model has a 3D wet mask, with certain deep cells masked due to topography? This is the typical case for z-coordinate ocean models, and we will need to solve it if we want to filter in 3D. My _hope_ is that xarray.apply_ufunc just does the right thing and provides the appropriate mask to each level. But I'm really not sure about that.

Has anyone (@NoraLoose, @ElizabethYankovsky) tested this case?

**Comment by NoraLoose** [comment]
@rabernat, I haven't tested the case of a 3D land mask yet. If @ElizabethYankovsky hasn't either, I can run a test case and report back.

**Comment by ElizabethYankovsky** [comment]
@NoraLoose I also haven't yet tested 3D data with a 3D land mask (only looked into filtering 2D fields)

**Comment by rabernat** [comment]
Ok, good to know. This is not urgent. Our priority right now is to finish the JAMES paper. Then we can turn our attention to stuff like this in the package. I just wanted to make of note of this before I forgot about it. 🙃

**Comment by iangrooms** [comment]
The POP data has a 3D land mask. Didn't you use it for the timing tests for the JAMES paper?

**Comment by NoraLoose** [comment]
@iangrooms, for the POP data I defined
```
wet_mask = xr.where(ds['KMT']>0,1,0)
```
as the wet mask. But `KMT` is only a 2D array, and has no z-dimension. Which variable is the 3D land mask you are referring to?

**Comment by iangrooms** [comment]
`KMT` is the land mask, even though it's 2D. If `KMT(i,j) < k` then layer `i,j,k` is on land (where the top layer/level is `k==1`). I suppose you could turn this into a 3D array with ones and zeros for land and wet; the POP developers had a nice idea to save space by only using a 2D array.

**Comment by NoraLoose** [comment]
It's been a while since I looked at `KMT`. But now that you are mentioning this, I vaguely remember this convention - thanks! 

To summarize, I used the surface 2D wet mask for filtering all 62 levels of the POP data, which was wrong. But I guess this doesn't matter now that the computational section is skipped. When I have time, I will have a look and see if filtering the POP data with the 3D wet mask works.
